### PR TITLE
Updated tests to support -UserAuth test resources switch to enable non-corpnet access.

### PR DESCRIPTION
**Pull request by LarryOsterman**
Recent policy changes to our tenant effectively block the use of ClientSecretCredential on non-corpnet joined machines. This PR makes the changes necessary to support the new -UserAuth switch which is needed to work around the policy changes.

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

File: sdk/core/perf/CMakeLists.txt [hunk 1]
```diff
@@ -51,7 +51,7 @@ endif()
 
 # make sure that users can consume the project as a library.
 add_library (Azure::Perf ALIAS azure-perf)
-target_link_libraries(azure-perf PUBLIC azure-core)
+target_link_libraries(azure-perf PUBLIC azure-core azure-identity)
```
Comment (gearama) [inline sdk/core/perf/CMakeLists.txt:54]: core depends on identity now  ?

Comment (LarryOsterman) [reply sdk/core/perf/CMakeLists.txt:54]: *perf* depends on identity. Perf is not a package.

Core depends on nothing.

**Review by RickWinter — APPROVED** [review]

**Review by gearama — APPROVED** [review]

**Pull request merged**